lib/compiler/test/beam_doc_SUITE.erl

```diff
@@ -0,0 +1,53 @@
+-module(beam_doc_SUITE).
+
+-export([all/0, simple_moduledoc/1]).
+-include_lib("common_test/include/ct.hrl").
+
+all() ->
+    [simple_moduledoc].
+
+
+
+simple_moduledoc(Config) when is_list(Config) ->
+    PrivDir = proplists:get_value(priv_dir, Config),
+
+    ModuleContent = <<"-module(moduledoc).
+
+                       -export([]).
+
+                       -moduledoc \"
+                       Moduledoc test module
+                       \".
+                       ">>,
+    {ModName, Filename} = create_file(PrivDir, ModuleContent),
+    %% positive test: checks that all works as expected
+    %% FileName = CreateFile("module_attr", ErlModName, ModuleFileContent),
+    {ok, ModName} = compile:file(Filename, [beam_docs]),
+    %% ct:pal("~p~n~p~n~p~n", [Result, ModuleAtom, Result =:= ModuleAtom]),
+
+
+    Mime = <<"text/markdown">>,
+    ModuleDoc = #{<<"en">> => <<"Moduledoc test module">>},
+    {ok, {docs_v1, _,_, Mime,ModuleDoc, _,_}} = code:get_doc(ModName),
+    ok.
+
+create_dir(Dir) ->
+  case file:make_dir(Dir) of
+    ok -> ok;
+    {error, eexist} -> ok;
+    _ -> error
+  end.
```

Comment:
Each new test run is started with a newly created private directory. Something is seriously wrong if the directory already exists ("let it fail").

```suggestion
  ok = file:make_dir(Dir).
```